app/controllers/HomeController.scala

```diff
@@ -61,6 +64,29 @@ class HomeController @Inject() (
   private val authenticate: ActionBuilder[UserRequest, AnyContent] =
     authJourney.authWithPersonalDetails
 
+  private def verifyOrBouncedHandler(
+    verifyOrBounced: Option[PaperlessMessages]
+  )(implicit request: UserRequest[_]): Option[String] = {
+    val absoluteUrl = configDecorator.pertaxFrontendHost + request.uri
+    verifyOrBounced match {
+      case Some(paperlessStatus: PaperlessStatusBounced)    =>
+        Some(
+          configDecorator.preferencedBouncedEmailLink(
+            tools.encryptAndEncode(absoluteUrl),
+            tools.encryptAndEncode(paperlessStatus.link)
+          )
+        )
+      case Some(paperlessStatus: PaperlessStatusUnverified) =>
+        Some(
+          configDecorator.preferencedBouncedEmailLink(
+            tools.encryptAndEncode(absoluteUrl),
+            tools.encryptAndEncode(paperlessStatus.link)
+          )
+        )
+      case _                                                => None
+    }
```

Comment:
In the future that could become a long list. Could we pass a List[Html] or similar? For now it will be only one element.